SomosCodear/codear-site
# Make the calendar a React component

### What does this PR do?

It ports the calendar component we built in Lilac using Lit-Element to React. The reason behind this is that we consider it to be very specific to this application and we were having some performance issues during render time.

Besides moving the calendar, I also took care of a few small things:

- Define the app breakpoints and base colors as constants.
- The calendar events are now links on both desktop and mobile.
- Add the web components polyfill for the things we are still using from Lilac.
- Replace the menu mobile background with a gradient instead of an image.
- Fix a Next SSR issue with `Link` adding "string callbacks" to wrapped web components:

```html
<lilac-button href="/conocenos" title="Conocé al equipo" secundario="true" onMouseEnter="e =&gt; {
    if (child.props &amp;&amp; typeof child.props.onMouseEnter === &#x27;function&#x27;) {
      child.props.onMouseEnter(e);
    }

    this.prefetch();
  }" onClick="e =&gt; {
    if (child.props &amp;&amp; typeof child.props.onClick === &#x27;function&#x27;) {
      child.props.onClick(e);
    }

    if (!e.defaultPrevented) {
      this.linkClicked(e);
    }
}">
```

### How should it be tested manually?

- The calendar on this branch should work the same as the one on `develop`, except for the fact of the events being links now.
- After rendering the home, open the source of the document (the one returned from the server, not the one on the devtools), you shouldn't see any JS callback.
- Open the menu on mobile, you shouldn't see the "blink" of the background being loaded.
- Try the app on MS Edge, it should render the web components (ignore the styling issue with grid for now).

### TODOs

- [ ] @joelalejandro I made the menu background a linear gradient from Disco to Jakarta as you requested, but the second diamond of the CoDeAr logo kind of disappears.
